### PR TITLE
Add enterprise compilation with subdir

### PR DESCRIPTION
**Pull request by rrooij**
Enable compilation for enterprise when building. Enterprise will be build if you clone the enterprise repo in a subdirectory. This works easier than using a different project directory entirely.

Furthermore, I improved the CI to make it reusable for enterprise as well. The Dockerfiles aren't duplicated anymore.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -101,7 +101,7 @@ docs-clean:
 $(TARGET): $(RUST_TARGET) $(PROLOG_FILES)
 	# Build the target and fail for errors and warnings. Ignore warnings
 	# having "qsave(strip_failed(..." that occur on macOS.
-	TERMINUSDB_ENTERPRISE=$(ENTERPRISE) $(SWIPL) -t 'main,halt.' -q -O -f src/bootstrap.pl 2>&1 | \
+	TERMINUSDB_ENTERPRISE=$(ENTERPRISE) $(SWIPL) -t 'main,halt.' -q -O -f src/bootstrap.pl
```
Comment (spl) [inline Makefile:104]: Is this change intentional? It seems like it would break the things mentioned in the `Makefile` comment just above.

File: src/load_paths.pl [hunk 2]
```diff
@@ -1,12 +1,18 @@
 :- dynamic user:file_search_path/2.
 :- multifile user:file_search_path/2.
 
+:- use_module(library(filesex)).
+
 :- set_prolog_flag(terminusdb_monolithic_module, true).
 
-add_terminus_home_path :-
+root_path(Path) :-
```
Comment (spl) [inline src/load_paths.pl:8]:
```suggestion
% The top-level directory of the repository.
top_level_directory(Path) :-
```

File: src/load_paths.pl [hunk 3]
```diff
@@ -1,12 +1,18 @@
...
     prolog_load_context(file, File),
-    file_directory_name(File, Dir),
-    asserta(user:file_search_path(terminus_home, Dir)).
+    relative_file_name(Path, File, '../').
```
Comment (spl) [inline src/load_paths.pl:10]:
```suggestion
    relative_file_name(Path, File, '..').
```

File: src/load_paths.pl [hunk 4]
```diff
@@ -1,12 +1,18 @@
...
+
+add_terminus_home_path :-
+    root_path(RootDir),
```
Comment (spl) [inline src/load_paths.pl:13]:
```suggestion
    top_level_directory(TopDir),
```

File: src/load_paths.pl [hunk 5]
```diff
@@ -1,12 +1,18 @@
...
+    atom_concat(RootDir, 'src', SrcDir),
```
Comment (spl) [inline src/load_paths.pl:14]:
```suggestion
    directory_file_path(TopDir, 'src', SrcDir),
```

File: src/load_paths.pl [hunk 6]
```diff
@@ -105,13 +111,13 @@
 :- add_rust_path.
 
 add_enterprise_path :-
-    user:file_search_path(terminus_home, Dir),
-    atom_concat(Dir, '/../../terminusdb-enterprise/prolog', Enterprise),
+    root_path(Dir),
```
Comment (spl) [inline src/load_paths.pl:114]:
```suggestion
    top_level_directory(TopDir),
```

File: src/load_paths.pl [hunk 7]
```diff
@@ -105,13 +111,13 @@
...
+    atom_concat(Dir, 'terminusdb-enterprise/prolog', Enterprise),
```
Comment (spl) [inline src/load_paths.pl:115]:
```suggestion
    directory_file_path(TopDir, 'terminusdb-enterprise/prolog', Enterprise),
```

File: src/load_paths.pl [hunk 8]
```diff
@@ -105,13 +111,13 @@
...
     asserta(user:file_search_path(enterprise, Enterprise)).
 
 add_enterprise_test_path :-
-    user:file_search_path(terminus_home, Dir),
-    atom_concat(Dir, '/../../terminusdb-enterprise/test', Enterprise),
+    root_path(Dir),
```
Comment (spl) [inline src/load_paths.pl:119]:
```suggestion
    top_level_directory(TopDir),
```

File: src/load_paths.pl [hunk 9]
```diff
@@ -105,13 +111,13 @@
...
+    atom_concat(Dir, 'terminusdb-enterprise/test', Enterprise),
```
Comment (spl) [inline src/load_paths.pl:120]:
```suggestion
    directory_file_path(TopDir, 'terminusdb-enterprise/test', Enterprise),
```

**Review by spl — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (rrooij) [reply Makefile:104]: WOOPS

**Comment by rrooij** [comment]
Thank you for reviewing. I think I fixed most of your comments. :smile:

**Review by spl — APPROVED** [review]
